### PR TITLE
Mm/add legacy react

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slice-machine/server/src/api/simulator.ts [hunk 1]
```diff
@@ -38,20 +65,8 @@ export default async function handler(
...
-          Utils.CONSTS.NEXT_SLICEZONE,
-          Utils.CONSTS.PRISMIC_REACT_PACKAGE_NAME,
-        ]
-      : [
-          Utils.CONSTS.SLICE_SIMULATOR_VUE,
-          Utils.CONSTS.NUXT_SM,
-          Utils.CONSTS.VUE_SLICEZONE,
-          Utils.CONSTS.NUXT_PRISMIC,
-        ];
+  // TODO: fix this for legacy frameworks, and svelte?
```
Comment (Miniplop) [inline packages/slice-machine/server/src/api/simulator.ts]: don't get it, svelte and other frameworks doesn't support the simulator why we should handle them at this point ?

**Review by Miniplop — CHANGES_REQUESTED** [review]
Two questions : 
-  Why creating the "previous-react" framework ? Do we want to be retro compatible on this framework ? 
- I think we avoid introducing previous and legacy word for representing the same thing, so I think we should rename all the "legacy" word into "previous". It will help us to maintain and remove easily this feature.

File: packages/slice-machine/server/src/api/simulator.ts (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/server/src/api/simulator.ts]: good point, what about for legacy nuxt?

Comment (Miniplop) [reply packages/slice-machine/server/src/api/simulator.ts]: It's in another task to support it :)

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/Renderers/index.tsx [hunk 2]
```diff
@@ -1,13 +1,15 @@
 import toVue from "./vue";
 import toReact from "./react";
 import toSvelte from "./svelte";
+import toPreviousReact from "./previous-react";
 
 export const svelte = toSvelte;
 
 export const vue = toVue;
-export const react = toReact;
+export const react = toPreviousReact;
```
Comment (Miniplop) [inline packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/Renderers/index.tsx]: I think here you want to use to React as the new user will use the helpers lib :)

File: packages/core/src/utils/const.ts [hunk 3]
```diff
@@ -3,9 +3,11 @@ export const SCRIPT_VALUE = "start-slicemachine";
 export const INIT_COMMAND = "npx sm init";
 export const SM_PACKAGE_NAME = "slice-machine-ui";
 
-export const PRISMIC_REACT_PACKAGE_NAME = "prismic-reactjs";
+export const PREVIOUS_REACT_PACKAGE_NAME = "prismic-reactjs";
```
Comment (Miniplop) [inline packages/core/src/utils/const.ts:6]: Is this variable still used ?

Comment (MarcMcIntosh) [reply packages/core/src/utils/const.ts:6]: yes

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/Renderers/index.tsx (hunk 2 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/Renderers/index.tsx]: Then that would be the new version

Comment (Miniplop) [reply packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/Renderers/index.tsx]: yes that's what we want, no backward compatibility :)

**Review by Miniplop — APPROVED** [review]
LGTM

**Pull request merged**